### ISSUE TITLE
MultilineLambdaItParameter: false positive for one-line statements with a lambda argument

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#5504]
## Steps to Reproduce

```kotlin
@Nested
inner class `single parameter, single-line lambda` {
    @Test
    fun `does not report when statement has one-line lambda argument`() {
        val code = """
        fun f() {
            val digits = 1234.let {
                check(it > 0) { println(it) }
            }
        }
        """.trimIndent()
        val findings = subject.compileAndLintWithContext(env, code)
        assertThat(findings).isEmpty()
    }
}
```

## Context
https://github.com/detekt/detekt/pull/5397#issuecomment-1302776667

**Comment by TWiStErRob** [comment]
Agreed. I reproed using his code:
```kotlin
    @Test
    fun `example1`() {
        val code = """
            interface DeathStar {
                fun seekTieFighters(block: () -> Unit)
            }
            class ForegroundColorSpan(color: Int)
            fun List<DeathStar>?.f(colorLink: Int) {
                this?.map { it.seekTieFighters { ForegroundColorSpan(colorLink) } }
            }
        """.trimIndent()
        val findings = subject.compileAndLintWithContext(env, code)
        assertThat(findings).isEmpty()
    }
```
> Expecting empty but was: [CodeSmell(issue=Issue(id='MultilineLambdaItParameter', severity=Style, debt=5min), entity=Entity(name=f, signature=Test.kt${ it.seekTieFighters { ForegroundColorSpan(colorLink) } }, location=Location(source=6:15, text=160:217, file=Test.kt, filePath=FilePath(absolutePath=Test.kt, basePath=null, relativePath=null)), ktElement=LAMBDA_EXPRESSION), message=The implicit `it` should not be used in a multiline lambda. Consider giving your parameter a readable and descriptive name., metrics=[], references=[], severity=WARNING, id='MultilineLambdaItParameter')]
	at io.gitlab.arturbosch.detekt.rules.style.MultilineLambdaItParameterSpec.example1(MultilineLambdaItParameterSpec.kt:241)

in this one there's only one `it`, could be simplified further though.

**Comment by TWiStErRob** [comment]
Example 1 is covered in #5504

---

Example 2: I agree that this is simple enough (more on this later), but I guess maybe the rule needs to be smarter, and either consider this style as a "single line", or detect that the receiver is a simple name. Because it might as well be `foo().bar.baz().z { fux(it) }?.lam?.da?.also { ...`, in which case naming the `it` is paramount. Again, a separate issue will help to discuss this. Did you open https://github.com/detekt/detekt/issues/5503 as a generalization of this example? Here's a repro:

```kotlin
    @Test
    fun `example2`() {
        val code = """
            interface TextView {
                var isVisible: Boolean
                var text: String
            }
            fun phrase(text: String, b1: Boolean, b2: Boolean, block: () -> String?): String = TODO()
            fun f(textView: TextView, text: String?) {
                textView.isVisible = text?.also {
                    textView.text = phrase(it, false, false) { null }
                } != null
            }
        """.trimIndent()
        val findings = subject.compileAndLintWithContext(env, code)
        assertThat(findings).isEmpty()
    }
```

---

Now, as for "simple enough". I strongly believe that when a Detekt (or any inspection / static analysis tool) flags a line of code, it is for a reason. But we should never take the suggestions literally as the rules are rules, but the solution might be something totally different. For example in your second example, it took me about a 30 seconds to decypher the data flow and nullabilities, which is unnecessarily long. The code could be way simpler:
```kotlin
val somekindOfPhrase = phrase(text, false, false) { null } // name based on those (false, false, null) behavior flags
textView.isVisible = somekindOfPhrase != null
textView.text = somekindOfPhrase
```
ok, this is not a single statement, but we don't need to take _"in Kotlin, everything is an expression"_ literally, right? 🤓

Note that the two `TextView` interactions lend themselves really nicely for an extension `var TextView.textOrGone: CharSequence?`, which is common enough in my experience to warrant a util. At which point we're down to:
```kotlin
textView.textOrGone = phrase(text, false, false) { null } 
```
